lib/util/propTypes.js

```diff
@@ -121,29 +127,36 @@ module.exports = function propTypesInstructions(context, components, utils) {
     },
 
     ObjectTypeAnnotation(annotation, parentName, seen) {
-      let containsObjectTypeSpread = false;
+      let containsUnresolvedObjectTypeSpread = false;
       const containsIndexers = Boolean(annotation.indexers && annotation.indexers.length);
       const shapeTypeDefinition = {
         type: 'shape',
         children: {}
       };
       iterateProperties(context, annotation.properties, (childKey, childValue, propNode) => {
         const fullName = [parentName, childKey].join('.');
-        if (!childKey && !childValue) {
-          containsObjectTypeSpread = true;
-        } else {
+        if (childKey || childValue) {
           const types = buildTypeAnnotationDeclarationTypes(childValue, fullName, seen);
           types.fullName = fullName;
           types.name = childKey;
           types.node = propNode;
           types.isRequired = !childValue.optional;
           shapeTypeDefinition.children[childKey] = types;
         }
+      },
+      (spreadNode) => {
+        const key = getKeyValue(context, spreadNode);
+        const types = buildTypeAnnotationDeclarationTypes(spreadNode, key, seen);
+        if (!types.children) {
+          containsUnresolvedObjectTypeSpread = true;
+        } else {
+          Object.assign(shapeTypeDefinition, types.children);
+        }
       });
 
       // Mark if this shape has spread or an indexer. We will know to consider all props from this shape as having propTypes,
       // but still have the ability to detect unused children of this shape.
-      shapeTypeDefinition.containsSpread = containsObjectTypeSpread;
```

Comment:
this might be a breaking change for anyone using the utility from outside. can we keep both?

Reply:
@ljharb updated